CMakeLists.txt

```diff
@@ -0,0 +1,845 @@
+  message("Current compiler is ${CMAKE_CXX_COMPILER_ID}")
+  message(FATAL_ERROR "Missing compiler. Must be GNU, Clang, or MSVC")
+endif()
+
+############################################################
+# Check if should use gcc4's ABI
+set(gcc4_abi false)
+
+if ($ENV{RIPPLED_OLD_GCC_ABI})
+  set(gcc4_abi true)
+endif()
+
+if (is_gcc AND NOT gcc4_abi)
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5)
+    execute_process(COMMAND lsb_release -si OUTPUT_VARIABLE lsb)
+    string(STRIP ${lsb} lsb)
+    if (${lsb} STREQUAL "Ubuntu")
+      execute_process(COMMAND lsb_release -sr OUTPUT_VARIABLE lsb)
+      string(STRIP ${lsb} lsb)
+      if (${lsb} VERSION_LESS 15.1)
+        set(gcc4_abi true)
+      endif()
+    endif()
+  endif()
+endif()
+
+if (gcc4_abi)
+  add_definitions(-D_GLIBCXX_USE_CXX11_ABI=0)
+endif()
+
+############################################################
+
+include_directories(src src/beast src/beast/include src/beast/extras
+  src/soci/src src/soci/include)
+
+if (coverage)
+  add_compile_options(-fprofile-arcs -ftest-coverage)
+  append_flags(CMAKE_EXE_LINKER_FLAGS -fprofile-arcs -ftest-coverage)
+endif()
+
+if (profile)
+  add_compile_options(-p -pg)
+  append_flags(CMAKE_EXE_LINKER_FLAGS -p -pg)
+endif()
+
+############################################################
+
+if (NOT WIN32)
+  if (is_clang AND DEFINED ENV{CLANG_BOOST_ROOT})
+    set(BOOST_ROOT $ENV{CLANG_BOOST_ROOT})
+  endif()
+
+
+  set(Boost_USE_STATIC_LIBS on)
+  set(Boost_USE_MULTITHREADED on)
+  set(Boost_USE_STATIC_RUNTIME off)
+  find_package(Boost COMPONENTS
+    coroutine
+    context
+    date_time
+    filesystem
+    program_options
+    regex
+    system
+    thread)
+
+  if (Boost_FOUND)
+    include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
+  else()
+    message(FATAL_ERROR "Boost not found")
+  endif()
+endif()
+
+if (APPLE)
+  # swd TBD fixme
+  file(GLOB OPENSSL_ROOT_DIR /usr/local/Cellar/openssl/*)
+  # set(OPENSSL_ROOT_DIR /usr/local/Cellar/openssl)
+endif()
+
+if (WIN32)
+  if (DEFINED ENV{OPENSSL_ROOT})
+    include_directories($ENV{OPENSSL_ROOT}/include)
+    link_directories($ENV{OPENSSL_ROOT}/lib)
+  endif()
+else()
+  if (static)
+    set(tmp CMAKE_FIND_LIBRARY_SUFFIXES)
+    set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
+  endif()
+
+  find_package(OpenSSL)
+
+  if (static)
+    set(CMAKE_FIND_LIBRARY_SUFFIXES tmp)
+  endif()
+
+  if (OPENSSL_FOUND)
+    include_directories(${OPENSSL_INCLUDE_DIR})
+  else()
+    message(FATAL_ERROR "OpenSSL not found")
+  endif()
+  if (UNIX AND NOT APPLE AND ${OPENSSL_VERSION} VERSION_LESS 1.0.2)
+    message(FATAL_ERROR
+      "Your openssl is Version: ${OPENSSL_VERSION}, rippled requires 1.0.2 or better.")
+  endif()
+endif()
+
+if (WIN32)
+  if (DEFINED ENV{PROTOBUF_ROOT})
+    include_directories($ENV{PROTOBUF_ROOT}/src)
+    link_directories($ENV{PROTOBUF_ROOT}/src/.libs)
+  endif()
+
+  # Modified from FindProtobuf.cmake
+  FUNCTION(PROTOBUF_GENERATE_CPP SRCS HDRS PROTOFILES)
```

Comment:
I'm guessing the standard finder function doesn't work well on windows? Since most Find* modules are community-provided, they are not well vetted on all platforms. e.g.:  https://cmake.org/Bug/view.php?id=14833